Documentation/authentication.md

```diff
@@ -0,0 +1,164 @@
+# Authentication Guide of etcd v3
+
+## Overview
+
+Authentication was added in etcd 2.1. In the v3 API, the authentication feature, its API and user interface, was modified slightly. This guide will help users set up basic authentication in etcd v3.
+
+## Special Users and Roles
+
+There is one special user, `root`, and there are two special roles, `root` and `guest`.
+
+### User `root`
+
+User `root` must be created before auth can be activated. It must have the `root` role and is allowed to change anything inside etcd. The idea behind the `root` user is for administrative purposes: managing other ordinal users and roles.  In addition, the root role is required for other management functionalities including adding/removing/updating a node in a cluster, defragmentation, taking a snapshot, and checking endpoint health.
+
+### Role `root`
+
+Role `root` can be granted to any user other than root. Having access via the root role not only allows global read-write access (as was the case before 2.1) but allows modification of the authentication policy and all administrative things, like modifying the cluster membership.
+
+## Working with users
+
+The `user` subcommand for `etcdctl` handles all things having to do with user accounts.
+
+A listing of users can be found with
+
+```
+$ etcdctl user list
+```
+
+Creating a user is as easy as
+
+```
+$ etcdctl user add myusername
+```
+
+And there will be prompt for a new password. The password can be supplied from standard input when an option `--interactive=false` is given.
+
+Roles can be granted and revoked for a user with
+
+```
+$ etcdctl user grant-role myusername foo
+$ etcdctl user revoke-role myusername bar
+```
+
+We can look at this user with
+
+```
+$ etcdctl user get myusername
+```
+
+And the password for a user can be changed with
+
+```
+$ etcdctl user passwd myusername
+```
+
+Which will prompt again for a new password. The password can be supplied from standard input when an option `--interactive=false` is given.
+
+To delete an account, there's always
+```
+$ etcdctl user delete myusername
+```
+
+
+## Working with roles
+
+The `role` subcommand for `etcdctl` handles all things having to do with access controls for particular roles, as were granted to individual users.
+
+A listing of roles can be found with
+
+```
+$ etcdctl role list
+```
+
+A new role can be created with
+
+```
+$ etcdctl role add myrolename
+```
+
+A role has no password; we are merely defining a new set of access rights.
+
+Roles are granted access to a single key or a range of keys.
+
+The range can be specified in a form of [start-key, end-key) where start-key should be smaller than end-key in an alphabetical manner.
```

Comment:
s/in a form of/as an interval
s/should be smaller than end-key in an alphabetical manner/should be lexically less than end-key, ?